integration/kubernetes/confidential/sev.bats

```diff
@@ -302,240 +51,235 @@ setup_file() {
+  local kbs_ip="$(ip -o route get to 8.8.8.8 | sed -n 's/.*src \([0-9.]\+\).*/\1/p')"
+  local kbs_uri="${kbs_ip}:44444"
 
+  # SEV unencrypted service yaml generation
+  k8s_generate_service_yaml "${TEST_DIR}/sev-unencrypted.yaml" "${IMAGE_REPO}:unencrypted"
+  
+  # SEV encrypted service yaml generation
+  # SEV policy is 3 (default):
+  # - NODBG (1): Debugging of the guest is disallowed when set
+  # - NOKS (2): Sharing keys with other guests is disallowed when set
+  k8s_generate_service_yaml "${TEST_DIR}/sev-encrypted.yaml" "${IMAGE_REPO}:multi-arch-encrypted"
+  k8s_yaml_set_annotation "${TEST_DIR}/sev-encrypted.yaml" "io.katacontainers.config.pre_attestation.uri" "${kbs_uri}"
+  k8s_yaml_set_annotation "${TEST_DIR}/sev-encrypted.yaml" "io.katacontainers.config.sev.policy" "3"
+  
   # SEV-ES policy is 7:
   # - NODBG (1): Debugging of the guest is disallowed when set
   # - NOKS (2): Sharing keys with other guests is disallowed when set
   # - ES (4): SEV-ES is required when set
-  generate_service_yaml "encrypted-image-tests-es" "${IMAGE_REPO}:multi-arch-encrypted" "7"
+  k8s_generate_service_yaml "${TEST_DIR}/sev-es-encrypted.yaml" "${IMAGE_REPO}:multi-arch-encrypted"
+  k8s_yaml_set_annotation "${TEST_DIR}/sev-es-encrypted.yaml" "io.katacontainers.config.pre_attestation.uri" "${kbs_uri}"
+  k8s_yaml_set_annotation "${TEST_DIR}/sev-es-encrypted.yaml" "io.katacontainers.config.sev.policy" "7"
+}
+
+teardown_file() {
+  # Allow to not destroy the environment if you are developing/debugging tests
+  if [[ "${CI:-false}" == "false" && "${DEBUG:-}" == true ]]; then
+    echo "Leaving changes and created resources untouched"
+    return
+  fi
+
+  # Remove all k8s test services
+  k8s_delete_all
 
-  echo "SETUP FILE - COMPLETE"
-  echo "###############################################################################"
+  # Stop the simple-kbs
+  simple_kbs_stop
+
+  # Cleanup directories
+  esudo rm -rf "${SIMPLE_KBS_DIR}"
+  esudo rm -rf "${TEST_DIR}"
 }
 
 setup() {
-  # Remove the service/deployment/pod if it exists
-  echo "Deleting previous test pods..."
-  delete_pods
-
-  # Delete any previous data in the DB
-  mysql -u${KBS_DB_USER} -p${KBS_DB_PW} -h ${KBS_DB_HOST} -D ${KBS_DB} <<EOF
-    DELETE FROM secrets WHERE id = 10;
-    DELETE FROM keysets WHERE id = 10;
-    DELETE FROM policy WHERE id = 10;
-EOF
+  # Remove any previous k8s test services
+  echo "Deleting previous test services..."
+  k8s_delete_all
+
+  # Delete any previous data in the simple-kbs database
+  simple_kbs_delete_data
 }
 
-@test "$test_tag Test SEV unencrypted container launch success" {
+
+@test "${TEST_TAG} Test SEV unencrypted container launch success" {
   # Turn off pre-attestation. It is not necessary for an unencrypted image.
-  esudo sed -i 's/guest_pre_attestation = true/guest_pre_attestation = false/g' ${SEV_CONFIG}
+  esudo sed -i 's/guest_pre_attestation = true/guest_pre_attestation = false/g' ${SEV_CONFIG_FILE}
```

Comment:
Can't this be done via pod annotation (`io.katacontainers.config.pre_attestation.enabled: "false"`)?

Reply:
Changed this to use pod annotation, thanks!.